In [1]:
import os
import csv
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import category_encoders as ce
import collections
from sklearn import preprocessing
from sklearn.preprocessing import MinMaxScaler
from sklearn.cluster import KMeans
from sklearn.decomposition import PCA

In [2]:
ds = pd.read_csv(r'C:\Users\Sofia\OneDrive - NOVAIMS\Nova IMS\Mestrado\2º semestre\Business Cases\BC2\dataset.csv', sep=";")
ds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 111733 entries, 0 to 111732
Data columns (total 29 columns):
ID                      111733 non-null int64
Nationality             111733 non-null object
Age                     107561 non-null float64
DaysSinceCreation       111733 non-null int64
NameHash                111733 non-null object
DocIDHash               110732 non-null object
AverageLeadTime         111733 non-null int64
LodgingRevenue          111733 non-null float64
OtherRevenue            111733 non-null float64
BookingsCanceled        111733 non-null int64
BookingsNoShowed        111733 non-null int64
BookingsCheckedIn       111733 non-null int64
PersonsNights           111733 non-null int64
RoomNights              111733 non-null int64
DistributionChannel     111733 non-null object
MarketSegment           111733 non-null object
SRHighFloor             111733 non-null int64
SRLowFloor              111733 non-null int64
SRAccessibleRoom        111733 non-null int64
SRMe

In [3]:
# Summary statistics for all variables
summary=ds.describe(exclude=[np.object])   # exclude only objects, that decrease readability and do not bring information
summary=summary.transpose()  # transpose the summary for easier reading
summary.head(len(summary))

count          mean           std   min      25%  \
ID                    111733.0  55867.000000  32254.683151   1.0  27934.0   
Age                   107561.0     45.639191     17.244952 -10.0     33.0   
DaysSinceCreation     111733.0    595.026599    374.657382  36.0    288.0   
AverageLeadTime       111733.0     60.833147     85.115320  -1.0      0.0   
LodgingRevenue        111733.0    283.851283    379.131556   0.0      0.0   
OtherRevenue          111733.0     64.682802    123.580715   0.0      0.0   
BookingsCanceled      111733.0      0.002282      0.080631   0.0      0.0   
BookingsNoShowed      111733.0      0.000600      0.028217   0.0      0.0   
BookingsCheckedIn     111733.0      0.737607      0.730889   0.0      0.0   
PersonsNights         111733.0      4.328318      4.630739   0.0      0.0   
RoomNights            111733.0      2.203825      2.301637   0.0      0.0   
SRHighFloor           111733.0      0.042512      0.201755   0.0      0.0   
SRLowFloor            111733.0      0.001307      0.036125   0.0      0.0   
SRAccessibleRoom      111733.0      0.000224      0.014957   0.0      0.0   
SRMediumFloor         111733.0      0.000770      0.027733   0.0      0.0   
SRBathtub             111733.0      0.003132      0.055881   0.0      0.0   
SRShower              111733.0      0.001629      0.040327   0.0      0.0   
SRCrib                111733.0      0.016181      0.126173   0.0      0.0   
SRKingSizeBed         111733.0      0.363268      0.480943   0.0      0.0   
SRTwinBed             111733.0      0.156811      0.363624   0.0      0.0   
SRNearElevator        111733.0      0.000331      0.018195   0.0      0.0   
SRAwayFromElevator    111733.0      0.003598      0.059874   0.0      0.0   
SRNoAlcoholInMiniBar  111733.0      0.000197      0.014031   0.0      0.0   
SRQuietRoom           111733.0      0.087718      0.282886   0.0      0.0   

                          50%      75%        max  
ID                    55867.0  83800.0  111733.00  
Age                      47.0     58.0     123.00  
DaysSinceCreation       522.0    889.0    1385.00  
AverageLeadTime          21.0     95.0     588.00  
LodgingRevenue          208.0    393.3   21781.00  
OtherRevenue             31.0     84.0    8859.25  
BookingsCanceled          0.0      0.0      15.00  
BookingsNoShowed          0.0      0.0       3.00  
BookingsCheckedIn         1.0      1.0      76.00  
PersonsNights             4.0      6.0     116.00  
RoomNights                2.0      3.0     185.00  
SRHighFloor               0.0      0.0       1.00  
SRLowFloor                0.0      0.0       1.00  
SRAccessibleRoom          0.0      0.0       1.00  
SRMediumFloor             0.0      0.0       1.00  
SRBathtub                 0.0      0.0       1.00  
SRShower                  0.0      0.0       1.00  
SRCrib                    0.0      0.0       1.00  
SRKingSizeBed             0.0      1.0       1.00  
SRTwinBed                 0.0      0.0       1.00  
SRNearElevator            0.0      0.0       1.00  
SRAwayFromElevator        0.0      0.0       1.00  
SRNoAlcoholInMiniBar      0.0      0.0       1.00  
SRQuietRoom               0.0      0.0       1.00

Questões gerais:
- Faz sentido analisarmos pessoas que não têm qualquer booking feito? Acho que não... têm valor de 0 em quase todas as variáveis.
- Quando é que foram recolhidos os dados? Data
- O que são os GDS Systems? É um distribution channel que junta os voos aos hotéis?
- A informação que temos na altura do booking varia consoante o tipo de channel?

Hash questions:
- É normal algumas pessoas com bookings feitos não terem documento identificação associado? Ser null...
- Pode acontecer que uma empresa faça bookings para os seus colaboradores e o DocIDHash seja igual para todos eles? Há tuplos com o mesmo Doc ID, mas com idades muito diferentes.
- Há um DocIDHash com 3000 observações com idades diferentes e muitos nulls na age, e tem todo o tipo de distribution channel. O que pode querer dizer?
No caso acima não faz sentido fazer merge porque claramente os bookings não dizem respeito todos à mesma pessoa... mas como controlamos isto? -> quando podemos ou não fazer merge.

Variáveis questões:
- O que fazemos com pessoas menores de idade? Que razões existem para eles estarem na nossa base de dados?
- As variáveis binárias são uma média das estadias ou têm a ver com a última estadia? O que fazemos quando fazemos merge de linhas?
- Avg lead time of -1? Means something or is an error? O empregado alterou os dados após a reserva, para corrigir algo?

1as conclusões das estatísticas descritivas:
- Age tem valor mínimo de -10 e máximo de 123 - coherence check
- Avg lead time of -1. Há alguma diferença entre o quartil dos 75% e o valor máximo e a média - possíveis outliers
- Lodging revenue - claros outliers
- Other revenue - claros outliers
- bookings cancelled - check for outliers
- boockings checked in - check for outliers
- persons nights - outliers
- room nights - outliers

Possíveis sugestões de implementação:
- Criar ficha cliente -> para beneficiar os clientes recorrentes.
- Guardar o nome da empresa quando é corporate.

In [4]:
ds[(ds.duplicated(subset=['DocIDHash']))].DocIDHash.value_counts()

0xE3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855    3031
0xA486FBACF4B4E5537B026743E3FDFE571D716839E758236F42950A61FE6B922B      30
0x2B17E9D2CCEF2EA0FE752EE345BEDFB06741FFC8ECECF45D6BBDBAF9A274FF52      23
0x469CF1F9CF8C790FFA5AD3F484F2938CBEFF6435BCFD734F687EC6D1E968F076      14
0x2A14D03A4827C67E0D39408F103DB417AD496DCE6158F8309E6281185C042003      13
                                                                      ... 
0xC2E9F5B7264AF770DB7E78D42C4A566BDAEDA6790C4A2CADD6CDBA30A22B41B1       1
0x50A6908B71943FBF414E6760C7696D61A58CFB29BB55C8392B32CD8E816E2BC6       1
0x6E08496D8485B95A699EDF39A4E5D3D7E6ACA349A23DC2FBB1ACB55DBF8075D3       1
0x6AF1D1958AF5DAEFE05DC55C0224231A5D9DE50F90A8B67B953BCEAA2C0755D0       1
0xCFE5F233866148F95D09F74B1449008412516CAA14380D00D917218B501BE75C       1
Name: DocIDHash, Length: 3085, dtype: int64

In [5]:
a=ds[(ds.duplicated(subset=['DocIDHash']))]

In [6]:
a[(a.DocIDHash=='0xE3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855') & (~a.Age.isnull())]

ID Nationality   Age  DaysSinceCreation  \
744        745         PRT  40.0               1359   
842        843         PRT  48.0               1355   
907        908         PRT  33.0               1352   
1002      1003         PRT  31.0               1348   
1003      1004         PRT  30.0               1348   
...        ...         ...   ...                ...   
103345  103346         PRT  23.0                103   
106428  106429         PRT  10.0                 78   
107858  107859         PRT   6.0                 67   
108450  108451         PRT  56.0                 62   
108968  108969         PRT  54.0                 58   

                                                 NameHash  \
744     0xDE75217D31BF3A07C97BC70DF817FCBA6CC5AA29C153...   
842     0x927BA21A63552D1FC65670C559845E2B3C19BCC55B18...   
907     0x3CA9DDAB19D8F50A425B14FAB6BA283701AB9B5A651C...   
1002    0x13B95955F49C8E660F3B7639FDAB4BB75A495493A5A2...   
1003    0x155357372FF051877CBE53E7A357935957CB3F22F8D7...   
...                                                   ...   
103345  0x3B383C03E88F6AFE5508D1E75BBDCB6FE13548B1BDA9...   
106428  0x8C93F42F7CE50F07E1B226CDD2111A5413903FA1EC73...   
107858  0x151FA6F1DCAD61BDCEF01D315AE43293597CE08B0F10...   
108450  0xD8F7D032DC3955A24CC5394935C8EDC4D06B77812C8A...   
108968  0x7C17DE4E2E27279DA7CD50C2A52030A413E057D34F1D...   

                                                DocIDHash  AverageLeadTime  \
744     0xE3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B...              103   
842     0xE3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B...               17   
907     0xE3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B...                3   
1002    0xE3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B...               22   
1003    0xE3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B...                0   
...                                                   ...              ...   
103345  0xE3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B...                1   
106428  0xE3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B...                0   
107858  0xE3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B...                0   
108450  0xE3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B...              170   
108968  0xE3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B...              166   

        LodgingRevenue  OtherRevenue  BookingsCanceled  ...  SRMediumFloor  \
744              388.0         120.0                 0  ...              0   
842              177.0          77.0                 0  ...              0   
907               73.0           7.0                 0  ...              0   
1002             172.8          42.0                 0  ...              0   
1003               0.0           0.0                 0  ...              0   
...                ...           ...               ...  ...            ...   
103345           116.0          16.0                 0  ...              0   
106428             0.0           0.0                 0  ...              0   
107858             0.0           0.0                 0  ...              0   
108450           314.0          32.0                 0  ...              0   
108968           938.7          28.0                 0  ...              0   

        SRBathtub  SRShower  SRCrib SRKingSizeBed SRTwinBed  SRNearElevator  \
744             0         0       0             0         0               0   
842             0         0       0             0         0               0   
907             0         0       0             0         0               0   
1002            0         0       0             1         0               0   
1003            0         0       0             1         0               0   
...           ...       ...     ...           ...       ...             ...   
103345          0         0       0             0         1               0   
106428          0         0       0             0         0               0   
107858          0         0      

In [9]:
ds[(ds.duplicated(subset=['DocIDHash']))].DocIDHash.value_counts()

0xE3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855    3031
0xA486FBACF4B4E5537B026743E3FDFE571D716839E758236F42950A61FE6B922B      30
0x2B17E9D2CCEF2EA0FE752EE345BEDFB06741FFC8ECECF45D6BBDBAF9A274FF52      23
0x469CF1F9CF8C790FFA5AD3F484F2938CBEFF6435BCFD734F687EC6D1E968F076      14
0x2A14D03A4827C67E0D39408F103DB417AD496DCE6158F8309E6281185C042003      13
                                                                      ... 
0xC2E9F5B7264AF770DB7E78D42C4A566BDAEDA6790C4A2CADD6CDBA30A22B41B1       1
0x50A6908B71943FBF414E6760C7696D61A58CFB29BB55C8392B32CD8E816E2BC6       1
0x6E08496D8485B95A699EDF39A4E5D3D7E6ACA349A23DC2FBB1ACB55DBF8075D3       1
0x6AF1D1958AF5DAEFE05DC55C0224231A5D9DE50F90A8B67B953BCEAA2C0755D0       1
0xCFE5F233866148F95D09F74B1449008412516CAA14380D00D917218B501BE75C       1
Name: DocIDHash, Length: 3085, dtype: int64

In [10]:
ds.DistributionChannel.unique()

array(['Corporate', 'Travel Agent/Operator', 'Direct', 'GDS Systems'],
      dtype=object)

In [12]:
ds[ds.DistributionChannel== 'Corporate'].duplicated(subset=['DocIDHash'])

0         False
374       False
378       False
402       False
405       False
          ...  
111143    False
111225     True
111226     True
111515    False
111601    False
Length: 3135, dtype: bool

In [13]:
ds[(ds.duplicated(subset=['DocIDHash']))&(ds.DistributionChannel== 'Corporate')].DocIDHash.value_counts()

0xE3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855    275
0xA486FBACF4B4E5537B026743E3FDFE571D716839E758236F42950A61FE6B922B     30
0x2B17E9D2CCEF2EA0FE752EE345BEDFB06741FFC8ECECF45D6BBDBAF9A274FF52     22
0x3856085146F7BC27BD07BFC4CA1991ED4E65E179D7BDB7DBBA7E32620809C799     11
0x9220D336F2DDD7B68F5066878889C7637EE28924B249F968F5EC82D895B108A7     11
                                                                     ... 
0xDE497DBCD45593B89D76F9B97CB1B9F3CF9DD677A16998FD35FFB2951A3E8CFD      1
0xD03C2CC7EB9FE7EB4FD0129C5AAAC23DE56B5124E2D937FF79FF4DDB949C521C      1
0x33B0972B52184C63AF39720B9DD68309776CF43398A8ED97657EFEDC095377C3      1
0xD30CB68082161621A24A5D86CADD8A6BD26814C7F65E49913D1B2D8C1D40B687      1
0x1AFF5611ED6521CEA7D3CA7A2A626B4F2C6AC53799534C00B7A2FD480A439E56      1
Name: DocIDHash, Length: 358, dtype: int64

In [14]:
ds['Total_Bookings']=ds.BookingsCanceled + ds.BookingsNoShowed + ds.BookingsCheckedIn

In [20]:
ds['Total_Revenue']=ds.LodgingRevenue + ds.OtherRevenue

In [15]:
ds.Total_Bookings

0         3
1         1
2         0
3         1
4         0
         ..
111728    0
111729    1
111730    0
111731    1
111732    0
Name: Total_Bookings, Length: 111733, dtype: int64

In [24]:
total = ds[(ds.Total_Bookings==0)]
total

ID Nationality   Age  DaysSinceCreation  \
2            3         DEU  32.0               1385   
4            5         FRA  52.0               1385   
6            7         JPN  50.0               1385   
8            9         FRA  43.0               1385   
10          11         IRL  28.0               1385   
...        ...         ...   ...                ...   
111723  111724         ITA  56.0                 37   
111725  111726         PAN  60.0                 37   
111728  111729         DEU  31.0                 36   
111730  111731         BRA  37.0                 36   
111732  111733         DEU  48.0                 36   

                                                 NameHash  \
2       0xDA46E62F66936284DF2844EC4FC542D0DAD780C0EE0C...   
4       0xD2E3D5BFCA141865669F98D64CDA85AD04DEFF47F8A0...   
6       0x94DB830C90A6DA2331968CFC9448AB9A3CE07D7CFEDD...   
8       0x44BB41EF2D87698E75B6FBB77A8815BF48DAA912C140...   
10      0x606584F7B9D6A1905BF9B7ED7B7EBE5DF17DF22FECDF...   
...                                                   ...   
111723  0x82277FA4AD074DA9786E64E977C94FEF1C2E47CD6212...   
111725  0xAA3529E305173010BADE927C7015FA091C3883D257E9...   
111728  0x2F59F0E86596B861C3303585FDB7F090B1B2D70442C5...   
111730  0xEF5F6AAA6120F2AC49179E16FD12477C243A8F4FC504...   
111732  0xE52EE501843E3108F20BB8F2EC180D8654DF1424EFA2...   

                                                DocIDHash  AverageLeadTime  \
2       0x27F5DF762CCDA622C752CCDA45794923BED9F1B66300...                0   
4       0x42BDEE0E05A9441C94147076EDDCC47E604DA5447DD4...                0   
6       0x47E5E4B21585F1FD956C768E730604241B380EDFEA68...                0   
8       0x6C456E45A78A20BC794137AE326A81D587B6528B3944...                0   
10      0xCA46CD508DB3DB2085374141177E07ECEC4B724AE41C...                0   
...                                                   ...              ...   
111723  0x5ED6055421E857EA3496D4C75FA2237ACD353DB1B64B...                0   
111725  0xDC47997245A7B9ACF9D4974149D2100CEB00981A2F03...                0   
111728  0x86AC02B9393E7F58354BA4193D792C07DB91D4BE27C7...                0   
111730  0x74B9B5A554F5F660371E56D4CE4C1D4C3DC80AE71D68...                0   
111732  0xD539E89183AB389864B7A24B3C1FBC1FE20A40ED02E7...                0   

        LodgingRevenue  OtherRevenue  BookingsCanceled  ...  SRShower  SRCrib  \
2                  0.0           0.0                 0  ...         0       0   
4                  0.0           0.0                 0  ...         0       0   
6                  0.0           0.0                 0  ...         0       0   
8                  0.0           0.0                 0  ...         0       0   
10                 0.0           0.0                 0  ...         0       0   
...                ...           ...               ...  ...       ...     ...   
111723             0.0           0.0                 0  ...         0       0   
111725             0.0           0.0                 0  ...         0       0   
111728             0.0           0.0                 0  ...         0       0   
111730             0.0           0.0                 0  ...         0       0   
111732             0.0           0.0                 0  ...         0       0   

        SRKingSizeBed  SRTwinBed SRNearElevator SRAwayFromElevator  \
2                   0          0              0                  0   
4                   0          0              0                  0   
6                   0          0              0                  0   
8                   1          0              0                  0   
10                  0          0              0                  0   
...               ...        ...            ...                ...   
111723              1          0              0                  0   
111725              0          1              1                  0   
111728              1          0              0                  0   
111730  

In [19]:
summary_total=total.describe(exclude=[np.object])   # exclude only objects, that decrease readability and do not bring information
summary_total=summary_total.transpose()  # transpose the summary for easier reading
summary_total.head(len(summary_total))

count          mean           std   min      25%  \
ID                    33197.0  75481.810796  23666.188706   3.0  59462.0   
Age                   32072.0     39.966139     20.205140  -6.0     24.0   
DaysSinceCreation     33197.0    365.488779    247.751981  36.0    176.0   
AverageLeadTime       33197.0      0.013947      2.015055   0.0      0.0   
LodgingRevenue        33197.0      0.007510      1.368273   0.0      0.0   
OtherRevenue          33197.0      0.000542      0.098792   0.0      0.0   
BookingsCanceled      33197.0      0.000000      0.000000   0.0      0.0   
BookingsNoShowed      33197.0      0.000000      0.000000   0.0      0.0   
BookingsCheckedIn     33197.0      0.000000      0.000000   0.0      0.0   
PersonsNights         33197.0      0.000000      0.000000   0.0      0.0   
RoomNights            33197.0      0.000000      0.000000   0.0      0.0   
SRHighFloor           33197.0      0.045667      0.208765   0.0      0.0   
SRLowFloor            33197.0      0.001476      0.038391   0.0      0.0   
SRAccessibleRoom      33197.0      0.000120      0.010976   0.0      0.0   
SRMediumFloor         33197.0      0.000693      0.026313   0.0      0.0   
SRBathtub             33197.0      0.003796      0.061492   0.0      0.0   
SRShower              33197.0      0.001566      0.039547   0.0      0.0   
SRCrib                33197.0      0.029551      0.169347   0.0      0.0   
SRKingSizeBed         33197.0      0.383438      0.486231   0.0      0.0   
SRTwinBed             33197.0      0.181673      0.385581   0.0      0.0   
SRNearElevator        33197.0      0.000271      0.016463   0.0      0.0   
SRAwayFromElevator    33197.0      0.003283      0.057208   0.0      0.0   
SRNoAlcoholInMiniBar  33197.0      0.000361      0.019009   0.0      0.0   
SRQuietRoom           33197.0      0.092026      0.289068   0.0      0.0   
Total_Bookings        33197.0      0.000000      0.000000   0.0      0.0   

                          50%      75%       max  
ID                    76182.0  94670.0  111733.0  
Age                      41.0     56.0     123.0  
DaysSinceCreation       369.0    493.0    1385.0  
AverageLeadTime           0.0      0.0     349.0  
LodgingRevenue            0.0      0.0     249.3  
OtherRevenue              0.0      0.0      18.0  
BookingsCanceled          0.0      0.0       0.0  
BookingsNoShowed          0.0      0.0       0.0  
BookingsCheckedIn         0.0      0.0       0.0  
PersonsNights             0.0      0.0       0.0  
RoomNights                0.0      0.0       0.0  
SRHighFloor               0.0      0.0       1.0  
SRLowFloor                0.0      0.0       1.0  
SRAccessibleRoom          0.0      0.0       1.0  
SRMediumFloor             0.0      0.0       1.0  
SRBathtub                 0.0      0.0       1.0  
SRShower                  0.0      0.0       1.0  
SRCrib                    0.0      0.0       1.0  
SRKingSizeBed             0.0      1.0       1.0  
SRTwinBed                 0.0      0.0       1.0  
SRNearElevator            0.0      0.0       1.0  
SRAwayFromElevator        0.0      0.0       1.0  
SRNoAlcoholInMiniBar      0.0      0.0       1.0  
SRQuietRoom               0.0      0.0       1.0  
Total_Bookings            0.0      0.0       0.0

In [23]:
total = ds[(ds.Total_Bookings==0)&(ds.Total_Revenue==0)]
total

ID Nationality   Age  DaysSinceCreation  \
2            3         DEU  32.0               1385   
4            5         FRA  52.0               1385   
6            7         JPN  50.0               1385   
8            9         FRA  43.0               1385   
10          11         IRL  28.0               1385   
...        ...         ...   ...                ...   
111723  111724         ITA  56.0                 37   
111725  111726         PAN  60.0                 37   
111728  111729         DEU  31.0                 36   
111730  111731         BRA  37.0                 36   
111732  111733         DEU  48.0                 36   

                                                 NameHash  \
2       0xDA46E62F66936284DF2844EC4FC542D0DAD780C0EE0C...   
4       0xD2E3D5BFCA141865669F98D64CDA85AD04DEFF47F8A0...   
6       0x94DB830C90A6DA2331968CFC9448AB9A3CE07D7CFEDD...   
8       0x44BB41EF2D87698E75B6FBB77A8815BF48DAA912C140...   
10      0x606584F7B9D6A1905BF9B7ED7B7EBE5DF17DF22FECDF...   
...                                                   ...   
111723  0x82277FA4AD074DA9786E64E977C94FEF1C2E47CD6212...   
111725  0xAA3529E305173010BADE927C7015FA091C3883D257E9...   
111728  0x2F59F0E86596B861C3303585FDB7F090B1B2D70442C5...   
111730  0xEF5F6AAA6120F2AC49179E16FD12477C243A8F4FC504...   
111732  0xE52EE501843E3108F20BB8F2EC180D8654DF1424EFA2...   

                                                DocIDHash  AverageLeadTime  \
2       0x27F5DF762CCDA622C752CCDA45794923BED9F1B66300...                0   
4       0x42BDEE0E05A9441C94147076EDDCC47E604DA5447DD4...                0   
6       0x47E5E4B21585F1FD956C768E730604241B380EDFEA68...                0   
8       0x6C456E45A78A20BC794137AE326A81D587B6528B3944...                0   
10      0xCA46CD508DB3DB2085374141177E07ECEC4B724AE41C...                0   
...                                                   ...              ...   
111723  0x5ED6055421E857EA3496D4C75FA2237ACD353DB1B64B...                0   
111725  0xDC47997245A7B9ACF9D4974149D2100CEB00981A2F03...                0   
111728  0x86AC02B9393E7F58354BA4193D792C07DB91D4BE27C7...                0   
111730  0x74B9B5A554F5F660371E56D4CE4C1D4C3DC80AE71D68...                0   
111732  0xD539E89183AB389864B7A24B3C1FBC1FE20A40ED02E7...                0   

        LodgingRevenue  OtherRevenue  BookingsCanceled  ...  SRShower  SRCrib  \
2                  0.0           0.0                 0  ...         0       0   
4                  0.0           0.0                 0  ...         0       0   
6                  0.0           0.0                 0  ...         0       0   
8                  0.0           0.0                 0  ...         0       0   
10                 0.0           0.0                 0  ...         0       0   
...                ...           ...               ...  ...       ...     ...   
111723             0.0           0.0                 0  ...         0       0   
111725             0.0           0.0                 0  ...         0       0   
111728             0.0           0.0                 0  ...         0       0   
111730             0.0           0.0                 0  ...         0       0   
111732             0.0           0.0                 0  ...         0       0   

        SRKingSizeBed  SRTwinBed SRNearElevator SRAwayFromElevator  \
2                   0          0              0                  0   
4                   0          0              0                  0   
6                   0          0              0                  0   
8                   1          0              0                  0   
10                  0          0              0                  0   
...               ...        ...            ...                ...   
111723              1          0              0                  0   
111725              0          1              1                  0   
111728              1          0              0                  0   
111730  

In [22]:
summary_total=total.describe(exclude=[np.object])   # exclude only objects, that decrease readability and do not bring information
summary_total=summary_total.transpose()  # transpose the summary for easier reading
summary_total.head(len(summary_total))

count          mean           std   min      25%  \
ID                    33196.0  75481.958459  23666.529882   3.0  59461.5   
Age                   32072.0     39.966139     20.205140  -6.0     24.0   
DaysSinceCreation     33196.0    365.487438    247.755592  36.0    176.0   
AverageLeadTime       33196.0      0.010513      1.915502   0.0      0.0   
LodgingRevenue        33196.0      0.000000      0.000000   0.0      0.0   
OtherRevenue          33196.0      0.000000      0.000000   0.0      0.0   
BookingsCanceled      33196.0      0.000000      0.000000   0.0      0.0   
BookingsNoShowed      33196.0      0.000000      0.000000   0.0      0.0   
BookingsCheckedIn     33196.0      0.000000      0.000000   0.0      0.0   
PersonsNights         33196.0      0.000000      0.000000   0.0      0.0   
RoomNights            33196.0      0.000000      0.000000   0.0      0.0   
SRHighFloor           33196.0      0.045668      0.208768   0.0      0.0   
SRLowFloor            33196.0      0.001476      0.038392   0.0      0.0   
SRAccessibleRoom      33196.0      0.000120      0.010977   0.0      0.0   
SRMediumFloor         33196.0      0.000693      0.026313   0.0      0.0   
SRBathtub             33196.0      0.003796      0.061493   0.0      0.0   
SRShower              33196.0      0.001566      0.039548   0.0      0.0   
SRCrib                33196.0      0.029552      0.169350   0.0      0.0   
SRKingSizeBed         33196.0      0.383450      0.486234   0.0      0.0   
SRTwinBed             33196.0      0.181679      0.385585   0.0      0.0   
SRNearElevator        33196.0      0.000271      0.016464   0.0      0.0   
SRAwayFromElevator    33196.0      0.003284      0.057209   0.0      0.0   
SRNoAlcoholInMiniBar  33196.0      0.000361      0.019010   0.0      0.0   
SRQuietRoom           33196.0      0.092029      0.289071   0.0      0.0   
Total_Bookings        33196.0      0.000000      0.000000   0.0      0.0   
Total_Revenue         33196.0      0.000000      0.000000   0.0      0.0   

                          50%      75%       max  
ID                    76183.0  94670.5  111733.0  
Age                      41.0     56.0     123.0  
DaysSinceCreation       369.0    493.0    1385.0  
AverageLeadTime           0.0      0.0     349.0  
LodgingRevenue            0.0      0.0       0.0  
OtherRevenue              0.0      0.0       0.0  
BookingsCanceled          0.0      0.0       0.0  
BookingsNoShowed          0.0      0.0       0.0  
BookingsCheckedIn         0.0      0.0       0.0  
PersonsNights             0.0      0.0       0.0  
RoomNights                0.0      0.0       0.0  
SRHighFloor               0.0      0.0       1.0  
SRLowFloor                0.0      0.0       1.0  
SRAccessibleRoom          0.0      0.0       1.0  
SRMediumFloor             0.0      0.0       1.0  
SRBathtub                 0.0      0.0       1.0  
SRShower                  0.0      0.0       1.0  
SRCrib                    0.0      0.0       1.0  
SRKingSizeBed             0.0      1.0       1.0  
SRTwinBed                 0.0      0.0       1.0  
SRNearElevator            0.0      0.0       1.0  
SRAwayFromElevator        0.0      0.0       1.0  
SRNoAlcoholInMiniBar      0.0      0.0       1.0  
SRQuietRoom               0.0      0.0       1.0  
Total_Bookings            0.0      0.0       0.0  
Total_Revenue             0.0      0.0       0.0